src/lib/Tab/LocationView/LocationsTab.php

```diff
@@ -112,12 +143,27 @@ public function getTemplateParameters(array $contextParameters = []): array
         $versionInfo = $content->getVersionInfo();
         $contentInfo = $versionInfo->getContentInfo();
         $locations = [];
+        $pagination = null;
 
         if ($contentInfo->published) {
-            $locationsDataset = $this->datasetFactory->locations();
-            $locationsDataset->load($contentInfo);
+            $currentPage = $this->requestStack->getCurrentRequest()->query->getInt(
+                self::PAGINATION_PARAM_NAME, 1
+            );
+
+            $locationQuery = new LocationQuery([
+                'filter' => new Query\Criterion\ContentId($contentInfo->id),
+            ]);
+
+            $pagination = new Pagerfanta(
+                new LocationSearchAdapter(
+                    $locationQuery,
+                    $this->searchService
+                )
+            );
 
-            $locations = $locationsDataset->getLocations();
+            $pagination->setMaxPerPage($this->defaultPaginationLimit);
```

Comment:
If this is SA-aware, it needs to be fetched on demand since the context is dynamic.

Reply:
I didn't see it being SA-aware

Reply:
You are using dynamic settings for it, so it is SA-aware.
https://github.com/ezsystems/ezplatform-admin-ui/pull/1735/files/968ecd76a9b3a82f547ea1287f4ab86f48a30de0#diff-10797a34251425cfff5006904962224a57f02b139b01a0f902ebf92a673d377eR18

Reply:
thanks for the info, this has been changed to use configResolver